internal/js/modules/k6/browser/common/frame.go

```diff
@@ -1082,13 +1082,20 @@ func (f *Frame) GetByRole(role string, opts *GetByRoleOptions) *Locator {
 
 // isQuotedText returns true if the string is a quoted string.
 // This is used to determine if the string will be used to match
-// a string instead of as a regex in the getBy* APIs.
+// a string instead of as a regex in the getBy* APIs. It handles
+// both single and double quotes.
 func isQuotedText(s string) bool {
-	return len(s) > 1 && s[0] == '\'' && s[len(s)-1] == '\''
+	switch {
```

Comment:
It seems you could simplify by removing len(s) > 0 and have a `if` statement in line outside with `len(s) < return false`

Reply:
You mean like:

```go
	if len(s) < 1 {
		return false
	}

	switch {
	case s[0] == '\'' && s[len(s)-1] == '\'':
		return true
	case s[0] == '"' && s[len(s)-1] == '"':
		return true
	}
	return false
```

If so, i've added it to the catch all [issue](https://github.com/grafana/k6/issues/4964) to work on later 👍